ARROW-4873: [C++] Clarify documentation about how to use external ARROW_PACKAGE_PREFIX while also using CONDA dependency resolution (#3901)
apache/arrow: cpp/CMakeLists.txt

  #----------------------------------------------------------------------
  # Thirdparty toolchain options

  # Determine how we will look for dependencies
  # * AUTO: Guess which packaging systems we're running in and pull the
  #   dependencies from there. Build any missing ones through the
  #   ExternalProject setup. This is the default unless the CONDA_PREFIX
  #   environment variable is set, in which case the CONDA method is used
  # * BUNDLED: Build all dependencies through CMake's ExternalProject facility.
  # * SYSTEM: Use CMake's find_package and find_library without any custom
  #   paths. If packages are on non-default locations, let the user indicate it

>   #   from the outside using the *_ROOT variables.  If your system packages are

Reviewer: nit:
```suggestion
  #   using the environment by setting *_ROOT variables.  If your system packages are
```

Author: Is this right? What the documentation currently says is to do:

```
cmake .. -DGTest_ROOT=$GTEST_PATH
```

https://github.com/apache/arrow/blob/master/cpp/cmake_modules/ThirdpartyToolchain.cmake#L86

Reviewer 2: Actually both are working in CMake 3.11+, before that only `-DGTest_ROOT=$GTEST_PATH` works.

  # * SYSTEM: Use CMake's find_package and find_library without any custom
  #   paths. If packages are on non-default locations, let the user indicate it
  #   from the outside using the *_ROOT variables.
  #   If your system packages are in general in a non-default location, or if
  #   you are using a non-standard toolchain, you can set a general default for
  #   the *_ROOT variables using ARROW_PACKAGE_PREFIX.
  # * CONDA: Same as system but set all *_ROOT variables to
  #   ENV{CONDA_PREFIX}. If this is run within an active conda environment,
  #   then ENV{CONDA_PREFIX} will be used for dependencies unless
  #   ARROW_DEPENDENCY_SOURCE is set explicitly to one of the other options

>   # * BREW: Use SYSTEM but search for select packages with brew.

Reviewer 2: I would actually like to keep this as the main strategy. Having conda packages and the build toolchain split is something that I would only advise to expert users since the compiler migration. With the new toolchain using system compilers, binutils, … is going to be a pain unless one is on a brand new system. I rather have CMake error out on missing conda packages than weird compilers as the binutils/compiler/ABI is not matching.

Author: Alright, I can write `-DARROW_DEPENDENCY_SOURCE=SYSTEM -DARROW_PACKAGE_PREFIX=$TOOLCHAIN` in the meantime
